probe-rs/src/config/target.rs

```diff
@@ -38,22 +38,23 @@ impl Target {
     /// Create a new target
     pub fn new(
         chip: &Chip,
+        cores: Vec<Core>,
         flash_algorithms: Vec<RawFlashAlgorithm>,
-        core_type: CoreType,
         source: TargetDescriptionSource,
     ) -> Target {
         Target {
             name: chip.name.clone(),
+            cores,
             flash_algorithms,
-            core_type,
             memory_map: chip.memory_map.clone(),
             source,
         }
     }
 
     /// Get the architectre of the target
     pub fn architecture(&self) -> Architecture {
-        match &self.core_type {
+        // TODO: what if they're different?
```

Comment:
Could it even?

Reply:
could be if the target yaml is wrong. It should probably be caught earlier, maybe in the "target validation" we once talked about

Reply:
There are designs with different cores: https://www.st.com/en/microcontrollers-microprocessors/stm32wb55cg.html
This is an M4 & M0+